polaris/hub/client.py

```diff
@@ -332,6 +333,33 @@ def get_dataset(self, owner: Union[str, HubOwner], name: str) -> Dataset:
 
         return Dataset(**response)
 
+    def read_zarr_file(self, owner: Union[str, HubOwner], name: str, path: str) -> zarr.hierarchy.Group:
+        """Read a Zarr file from a Polaris dataset
+
+        Args:
+            owner: Which Hub user or organization owns the artifact.
+            name: Name of the dataset.
+            path: Path to the Zarr file within the dataset.
+
+        Returns:
+            The Zarr object representing the dataset.
+        """
+        polaris_fs = PolarisFSFileSystem(
+            polaris_client=self,
+            dataset_owner=owner,
+            dataset_name=name,
+        )
+
+        try:
+            store = zarr.storage.FSStore(path, fs=polaris_fs)
+            return zarr.open(store, mode="r")
+        except TimeoutError as timeout_error:
```

Comment:
Looking at the HTTPX documentation, it appears `httpx` will throw a custom `TimeoutException`. See: https://www.python-httpx.org/exceptions/#the-exception-hierarchy